File: app/src/main/java/com/anandm/scanner/ScannerApp.java
package com.anandm.scanner;

import com.anandm.scanner.di.DaggerAppComponent;

import dagger.android.AndroidInjector;
import dagger.android.support.DaggerApplication;

public class ScannerApp extends DaggerApplication {

    @Override
    protected AndroidInjector<? extends DaggerApplication> applicationInjector() {
        return DaggerAppComponent.builder().application(this).build();
    }

}

File: app/src/main/java/com/anandm/scanner/ScannerApp.kt
package com.anandm.scanner

import com.anandm.scanner.di.DaggerAppComponent
import dagger.android.AndroidInjector
import dagger.android.support.DaggerApplication

class ScannerApp : DaggerApplication() {
    override fun applicationInjector(): AndroidInjector<out DaggerApplication> {
        return DaggerAppComponent.builder().application(this).build()
    }
}